packages/api/tests/endpoints/test-granules.js

```diff
@@ -3493,39 +3556,329 @@ test.serial('associateExecution (POST) returns Not Found if collectionId in payl
+    cumulus_id: newPgGranule.cumulus_id,
+  });
+  const translatedActualPgGranule = await translatePostgresGranuleToApiGranule({
+    knexOrTransaction: knex,
+    granulePgRecord: actualPgGranule,
+  });
+
+  const expectedGranule = {
+    ...newGranule,
+    error: {}, // This is a default value for no execution
+    published: false, // This is a default value
+    execution: apiGranule.execution, // This should not have changed
     status: 'completed',
+    timestamp: translatedActualPgGranule.timestamp,
+    updatedAt: translatedActualPgGranule.updatedAt,
+    createdAt: translatedActualPgGranule.createdAt,
   };
 
-  const expressRequest = {
-    params: {
-      granuleName: updatedGranule.granuleId,
-    },
-    body: updatedGranule,
-    testContext: {
-      knex,
-      esClient: fakeEsClient,
-    },
+  const updatedEsRecord = await t.context.esGranulesClient.get(
+    apiGranule.granuleId
+  );
+
+  // Files is always returned as '[]' by translator if none exist
+  t.deepEqual(
+    { ...translatedActualPgGranule },
+    { ...expectedGranule, files: [] }
+  );
+  t.deepEqual(updatedEsRecord, { ...expectedGranule, _id: updatedEsRecord._id });
+});
+
+test.serial('PUT creates a new granule in all data stores', async (t) => {
+  const {
+    collectionId,
+    collectionCumulusId,
+    createGranuleId,
+    knex,
+  } = t.context;
+
+  const granuleId = createGranuleId();
+  const newGranule = {
+    granuleId,
+    collectionId,
+    status: 'completed',
   };
-  const response = buildFakeExpressResponse();
 
-  put(expressRequest, response);
+  await request(app)
+    .put(`/granules/${t.context.collectionId}/${granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(newGranule)
+    .expect(201); // 201 should be expected for *create*
 
-  t.true(response.boom.badRequest.calledWithMatch('put method not implemented'));
+  const actualPgGranule = await t.context.granulePgModel.get(t.context.knex, {
+    collection_cumulus_id: collectionCumulusId,
+    granule_id: granuleId,
+  });
+  const translatedActualPgGranule = await translatePostgresGranuleToApiGranule({
+    knexOrTransaction: knex,
+    granulePgRecord: actualPgGranule,
+  });
+  const updatedEsRecord = await t.context.esGranulesClient.get(granuleId);
+
+  const expectedGranule = {
+    ...newGranule,
+    error: {}, // This is a default value for no execution
+    published: false, // This is a default value
+    timestamp: translatedActualPgGranule.timestamp,
+    updatedAt: translatedActualPgGranule.updatedAt,
+    createdAt: translatedActualPgGranule.createdAt,
+  };
+
+  // Files is always returned as '[]' via translator
+  t.deepEqual(
+    { ...translatedActualPgGranule },
+    { ...expectedGranule, files: [] }
+  );
+  t.deepEqual(updatedEsRecord, { ...expectedGranule, _id: updatedEsRecord._id });
+});
+
+test.serial('PUT utilizes the collectionId from the URI if one is not provided', async (t) => {
```

Comment:
This relates to an early comment. So the preference is to use the payload for the granuleId and collectionId? If those aren't supplied, use the URI? Could you briefly explain the thinking here just for my understanding?

Reply:
Discussed in huddle - the idea is to add values to the payload from the URI if they're missing as convenience to users.      Will update code and resubmit for review to do this for granuleId as well.